x/merkledb/intermediate_node_db.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package merkledb
+
+import (
+	"sync"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+var intermediateNodePrefixBytes = []byte(intermediateNodePrefix)
+
+const (
+	DefaultBufferLength       = 256
+	intermediateNodePrefix    = "intermediateNode"
```

Comment:
I think this should be as short as possible right? Feels like every prefix we use should be 1 byte here.